tests/cmsis_rtos_v1/src/mailq.c

```diff
@@ -38,24 +45,60 @@ void send_thread(void const *argument)
 	tx_ptr->data3 = MAIL1_DATA3;
 	status = osMailPut(mail_id, tx_ptr);
 	zassert_true(status == osOK, "osMailPut failure for mail1");
-	osDelay(100);
 
-	/* Prepare and send 2nd mail */
-	tx_ptr = osMailCAlloc(mail_id, osWaitForever);
-	zassert_true(tx_ptr != NULL, "Mail2 alloc failed");
+	/* Fill the queue with blocks of mails */
+	for (i = 0; i < Q_LEN; i++) {
+		tx_ptr = osMailAlloc(mail_id, osWaitForever);
+		zassert_true(tx_ptr != NULL, "Mail alloc failed");
+
+		tx_ptr->data1 = i;
+		tx_ptr->data2 = i+1;
+		tx_ptr->data3 = i+2;
+
+		status = osMailPut(mail_id, tx_ptr);
+		zassert_true(status == osOK,
+				"osMailPut failure for mail!");
+	}
+
+	/* Try allocating mail to a full queue immediately
+	 * before it is emptied out and assert failure
+	 */
+	tx_ptr = osMailAlloc(mail_id, 0);
+	zassert_true(tx_ptr == NULL, "Mail allocated. Something's wrong");
+
+	/* Try putting mail to a full queue within a duration
+	 * less than TIMEOUT, before the queue is emptied out
+	 */
+	tx_ptr = osMailAlloc(mail_id, TIMEOUT/2);
+	zassert_true(tx_ptr == NULL, "Mail allocated. Something's wrong");
+
+	/* Send another mail after the queue is emptied */
+	tx_ptr = osMailCAlloc(mail_id, TIMEOUT*2);
```

Comment:
Since there are both `osMailCAlloc` and `osMailAlloc` calls with three different timeouts to be covered, maybe you can call `osMailAlloc` when `i` is even, and `osMailCAlloc` when `i` is odd? We would hit all the cases then, improving code coverage, without writing redundant tests.

Reply:
Addressed. Thanks for pointing this out.